### PR TITLE
Serve contents of SD card through config server

### PR DESCRIPTION
**Pull request by opatut**
A very simple UI that displays file on the SD card and allows to download them. It looks like this:

image

### CODE REVIEW

File: OpenBikeSensorFirmware/configServer.cpp [hunk 1]
```diff
@@ -1065,6 +1069,90 @@ void startServer() {
...
+        child = file.openNextFile();
+      }
+
+      file.close();
+
+      html += "</ul>";
+      html += footer;
+      server.send(200, "text/html", html);
+
+      file.close();
```
Comment (amandel) [inline OpenBikeSensorFirmware/configServer.cpp:1121]: Is already closed 4 lines above

File: OpenBikeSensorFirmware/configServer.cpp [hunk 2]
```diff
@@ -1065,6 +1069,90 @@ void startServer() {
...
+    } else if (path.endsWith(".pdf")) {
+      dataType = "application/pdf";
+    } else if (path.endsWith(".zip")) {
+      dataType = "application/zip";
+    } else {
+      // arbitrary data
+      dataType = "application/octet-stream";
+    }
+
+    server.sendHeader("Content-Disposition", String("attachment; filename=\"") + file.name() + String("\""));
```
Comment (amandel) [inline OpenBikeSensorFirmware/configServer.cpp:1151]: Could set content-length?

**Review by amandel — APPROVED** [review]
LGTM

File: OpenBikeSensorFirmware/configServer.cpp (hunk 2 above)
Comment (opatut) [reply OpenBikeSensorFirmware/configServer.cpp:1151]: I think that's done by the `streamFile` below. At least, it works fine for me ;)

File: OpenBikeSensorFirmware/configServer.cpp (hunk 1 above)
Comment (opatut) [reply OpenBikeSensorFirmware/configServer.cpp:1121]: Yeah I should remove that. I tried reopening the file and only reading directories the first time, then files the second time, to get them sorted. But that somehow did not work. Forgot to remove this line from the second time.

File: OpenBikeSensorFirmware/configServer.cpp (hunk 2 above)
Comment (opatut) [reply OpenBikeSensorFirmware/configServer.cpp:1151]: image

Yes, it does :)

**Pull request merged**